### PR TITLE
Added ExecutionContext for passing to Hook methods

### PR DESCRIPTION
**Pull request by imonthercks**
This pull request was created based on a suggestion from @riju91 to split PR #124 so that changes to Lib could be built, tested and reviewed independently.

### CODE REVIEW

File: Lib/ExecutionContext.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
+// Copyright 2015 ThoughtWorks, Inc.
+
```
Comment (sriv) [inline Lib/ExecutionContext.cs]: The header is incorrect. This is from gauge-java. Please use the header from another gauge-csharp file.

Comment (imonthercks) [reply Lib/ExecutionContext.cs]: This has been corrected and pushed.  Thanks for catching this.  I will check the headers on my other pull request.

**Comment by imonthercks** [comment]
It appears the Travis build on Linux failed, but considering the only change was in a comment, I don't think it was related to my change.

**Comment by sriv** [comment]
cheers @imonthercks

**Pull request merged**